### PR TITLE
MCR-5334 Create Fetch and Create for OAuth

### PR DESCRIPTION
**Pull request by shamoons**
## Summary

This pull request introduces functionality for managing OAuth clients, including creating and fetching OAuth client credentials, integrating these capabilities into the existing GraphQL API, and adding comprehensive tests.

The most important changes are grouped into three areas:

### Backend Store Updates

- Added methods to the `Store` interface and `NewPostgresStore` for managing OAuth clients:
  - `createOAuthClient`
  - `listOAuthClients`
  - `getOAuthClientById`
  - `getOAuthClientByClientId`

### GraphQL Resolver and Schema Updates

- Implemented new GraphQL resolvers:
  - `createOauthClient` for creating OAuth client credentials.
  - `fetchOauthClients` for retrieving OAuth clients.
- Updated GraphQL schema with:
  - Types: `OauthClient`, `CreateOauthClientPayload`, `FetchOauthClientsPayload`
  - Inputs: `CreateOauthClientInput`, `FetchOauthClientsInput`

### Testing

- Added unit tests for `createOauthClient` resolver:
  - Covers valid input, defaults, permission checks, and error conditions.
- Added unit tests for `fetchOauthClients` resolver:
  - Covers list retrieval, filtering by ID, permission enforcement, and edge cases.

#### Related issues

- [MCR-5334](https://jiraent.cms.gov/browse/MCR-5334)

#### Screenshots

_N/A_

#### Test cases covered

- `createOauthClient.test.ts`:
  - Successful creation of OAuth clients
  - Enforcement of admin-only access
  - Input validation and fallback defaults
- `fetchOauthClients.test.ts`:
  - Fetching all clients and filtering by ID
  - Handling of unauthorized access
  - Error response validation

### CODE REVIEW

**Review by JasonLin0991 — COMMENTED** [review]
Lots of comments, but mainly just adding in error handling and logging, but it is blocking the merge.

I am curious about how we are creating Oauth clients, I thought we would be calling the Oauth server to generate our client data.

**Review by JasonLin0991 — APPROVED** [review]
Looks good. 

I left a comment about moving `callOauthLambdaToCreateCredentials` to the store function `createOAuthClient` either put the code inside that function or put the function in the same file `oauthClientStore` then calling it from `createOAuthClient`.

Approving, but would like to see that change.

File: services/app-api/src/resolvers/oauthClient/createOauthClient.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { v4 as uuidv4 } from 'uuid'
+import { randomBytes } from 'crypto'
+
+/**
+ * Generates a secure OAuth clientId and clientSecret.
+ * - clientId: uuid-based, prefixed with 'oauth-client-'
+ * - clientSecret: 64 random bytes, base64url encoded
+ * - grants: defaults to ['client_credentials'] if not provided
+ */
+async function callOauthLambdaToCreateCredentials(input: {
```
[the remark this answers is not in the document]
Comment (shamoons) [reply services/app-api/src/resolvers/oauthClient/createOauthClient.ts]: Done and done. Thanks for the callout!
